lib/tasks/base_path.rake

```diff
@@ -0,0 +1,24 @@
+namespace :base_path do
+  desc "Edits the base_path of a document, given its content_id and the new base_path"
+  task :edit, [:content_id, :base_path] => [:environment] do |_t, args|
+    content_id = args[:content_id]
+    base_path = args[:base_path]
+
+    begin
+      document = Document.find(content_id)
+    rescue DocumentFinder::RecordNotFound => e
+      puts "Error finding the document: #{e.inspect}"
```

Comment:
It looks like there are actually a few different exceptions which could be raised here:

```
irb(main):001:0> document = Document.find('')
...
GdsApi::TimedOutException (GdsApi::TimedOutException)

irb(main):003:0> document = Document.find('foo')
...
DocumentFinder::RecordNotFound (Document: foo)

irb(main):004:0> document = Document.find('foo bar baz')
...
GdsApi::InvalidUrl (GdsApi::InvalidUrl)
```

I'm not sure that catching the exception and displaying a message is any more useful than just letting the exception give a stack trace and terminate the task.

Reply:
Yes, I wasn't sure about this either, to be honest.
I followed the pattern that I saw already implemented in the code, but I'm usually a fan of letting things fail and dealing with the consequenses.
I'll change it.